xcwd.c

```diff
@@ -192,33 +192,29 @@ static processes_t getProcesses(void)
     p = malloc(sizeof(struct processes_s));
     struct kinfo_proc *kp;
     size_t len = 0;
-    int name[4] = { CTL_KERN, KERN_PROC, KERN_PROC_PROC, 0 };
-    sysctl(name, 4, NULL, &len, NULL, 0);
+    int name[6] = { CTL_KERN, KERN_PROC, KERN_PROC_UID, getuid(), sizeof(struct kinfo_proc), 0 };
+    sysctl(name, 6, NULL, &len, NULL, 0);
+    len += len/8; // some headroom
```

Comment:
Why do we need this?

Reply:
The number of processes can change between calls. This allows for some extra. Of course no guarantee.